### DISCUSSION TITLE
Is there a way to scroll down on a page until a certain element pops up ?

### DISCUSSION THREAD

**Discussion by elnatan2604** [seleniumbase/SeleniumBase#2367] [Q&A] [answered]
As I ask on the title, is there a way to do it?
I have an element that is not on the screen and I need to scroll down to it for be able to click on it.
I cant use normal scroll_to because the element isnt on the screen.

**Comment by mdmintz** [comment, marked as answer]
There are several scroll methods that you can work with:

```python
self.scroll_to(selector, by="css selector", timeout=None)
# Duplicates: self.scroll_to_element(selector, by="css selector")

self.slow_scroll_to(selector, by="css selector", timeout=None)
# Duplicates: self.slow_scroll_to_element(selector, by="css selector")

self.scroll_into_view(selector, by="css selector", timeout=None)

self.scroll_to_top()

self.scroll_to_bottom()
```

If the element that you want to interact with is already in the html, (but not visible yet), then you can use `self.js_click(selector)` to click on the hidden element. If that's not enough, you can create a loop where you call one of the scroll methods and then check for visibility before continuing the loop if the element wasn't visible yet. For example:

```python
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class MyTestClass(BaseCase):
    def test_base(self):
        self.open("URL")
        for i in range(20):
            self.execute_script("window.scrollTo(0, %s);" % (i * 100))
            if self.is_element_visible("SELECTOR"):
                break
            self.sleep(0.1)
```

**Reply by iamumairayub** [reply]
How do I use these methods when driver is created with `Driver()` class?

```
from seleniumbase import Driver

driver = Driver(
    uc=True, 
)
driver.get('https://www.amazon.co.uk/s?k=test&crid=16478HWXKS706&sprefix=%2Caps%2C571&ref=nb_sb_noss')
slow_scroll_to_element('a.s-pagination-item.s-pagination-next', by="css selector")
```

**Reply by mdmintz** [reply]
The `Driver()` format doesn't have all the methods of regular SeleniumBase when using the `SB()` or `BaseCase` formats. You would need to switch to one of the other formats to access that.

**Reply by elnatan2604** [reply]
I have an element but it is hidden.
Lets say I have this html code: "//div[@class='test']//input".
On my website I need to scroll down to it, wait for the website to finished download all the elements and then I can find it.
Is there a way to make my code scroll down (slowly) on the website until the element pops up?

**Reply by mdmintz** [reply]
Use a loop:

```python
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class MyTestClass(BaseCase):
    def test_base(self):
        self.open("URL")
        for i in range(20):
            self.execute_script("window.scrollTo(0, %s);" % (i * 100))
            if self.is_element_visible("SELECTOR"):
                break
            self.sleep(0.1)
```
